lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,485 @@
+        Output::Print(_u("EntryIntl_CurrencyDigits > returning (%d)\n"), minFracDigits);
+#endif
+
+        return minFracDigits;
+    }
+
+    template <typename Func>
+    HRESULT CreateFormatter(Func function, _In_z_ const char16 *languageTag, _In_ const charcount_t cch, _Out_ IPlatformAgnosticResource **resource)
+    {
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+
+        // Allocate memory for the UTF8 output buffer. Need 3 bytes for each (code point + null) to satisfy SAL.
+        const size_t inputLangTagUtf8SizeAllocated = AllocSizeMath::Mul(AllocSizeMath::Add(cch, 1), 3);
+        unsigned char *inputLangTagUtf8 = new unsigned char[inputLangTagUtf8SizeAllocated];
+        StringBufferAutoPtr<unsigned char> guard(inputLangTagUtf8);
+
+        if (!inputLangTagUtf8)
+        {
+            AssertOrFailFastMsg(false, "OOM: failed to allocate inputLangTagUtf8.");
+            return E_OUTOFMEMORY;
+        }
+
+        // explicitly zero the allocated array
+        for (size_t index = 0; index < inputLangTagUtf8SizeAllocated; ++index)
+        {
+            inputLangTagUtf8[index] = 0;
+        }
+
+        utf8::EncodeIntoAndNullTerminate(inputLangTagUtf8, languageTag, cch);
+
+        char *localeName = reinterpret_cast<char *>(inputLangTagUtf8);
+        auto locale = icu::Locale::createFromName(localeName);
+
+        icu::NumberFormat *nf = function(locale, error);
+
+        if (U_FAILURE(error))
+        {
+            AssertMsg(false, "Creating icu::NumberFormat failed");
+            return E_INVALIDARG;
+        }
+
+        auto *formatterResource = new PlatformAgnosticIntlObject<icu::NumberFormat>(nf);
+        if (!formatterResource)
+        {
+            return E_OUTOFMEMORY;
+        }
+
+        *resource = formatterResource;
+        return S_OK;
+    }
+
+    HRESULT CreateNumberFormatter(_In_z_ const char16 *languageTag, _In_ const charcount_t cch, _Out_ IPlatformAgnosticResource **resource)
+    {
+        //return CreateFormatter(icu::NumberFormat::createInstance, languageTag, cch, resource);
+        return CreateFormatter(
+            [](icu::Locale &locale, UErrorCode &error) { return icu::NumberFormat::createInstance(locale, error); },
+            languageTag, cch, resource);
+    }
+
+    HRESULT CreatePercentFormatter(_In_z_ const char16 *languageTag, _In_ const charcount_t cch, _Out_ IPlatformAgnosticResource **resource)
+    {
+        return CreateFormatter(
+            [](icu::Locale &locale, UErrorCode &error) { return icu::NumberFormat::createPercentInstance(locale, error); },
+            languageTag, cch, resource);
+    }
+
+    HRESULT CreateCurrencyFormatter(_In_z_ const char16 *languageTag, _In_ const charcount_t cch,
+        _In_z_ const char16 *currencyCode, _In_ const uint32 currencyDisplay, _Out_ IPlatformAgnosticResource **resource)
+    {
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+
+        unsigned char *inputLangTagUtf8 = nullptr;
+
+        {
```

Comment:
Intentional block?

Reply:
Yeah I had some dumb idea about limiting variable scope (in this case only inputLangTagUtf8SizeAllocated). I might have had an AutoPtr in mind at some point. Didn't use it. I'll remove the block.